packages/typespec-azure-resource-manager/lib/decorators.tsp

```diff
@@ -186,3 +186,21 @@ extern dec armCommonTypesVersion(
   target: Namespace | EnumMember,
   version: valueof string | EnumMember
 );
+
+/**
+ * This decorator is used on Azure Resource Manager resources that are not based on
+ * Azure.ResourceManager common types.
+ *
+ * @param propertiesType: The type of the resource properties.
+ */
+extern dec armBuiltInResource(target: Model, propertiesType: Model);
```

Comment:
Are those decorators supposed to be private?

Reply:
No,  A spec author could use these to more easily create additional path segments in the resource URI, so they should be public